### PR TITLE
#1399. [Records] Primitive equality tests added

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

File: LanguageFeatures/Records/primitive_equality_A01_t03.dart [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  const C();
+}
+
+main() {
+  const m1 = {const (1, "2", String, C(), Object(), Symbol('foo')): 1};
+  const m2 = {
+    (1, "2", String, const C(), const Object(), const Symbol('foo')): 1};
+  const m3 = {
+    const (one: 1, two: "2", three: String, four: C(), five: Object(),
+      six: Symbol('foo'): 2)
```
Comment (eernstg) [inline LanguageFeatures/Records/primitive_equality_A01_t03.dart:33]: Maybe?:
```suggestion
      six: Symbol('foo')): 2
```

File: LanguageFeatures/Records/primitive_equality_A01_t03.dart [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  };
+  const m4 = {
+    (one: 1, two: "2", three: String, four: const C(), five: const Object(),
+      six: const Symbol('foo'): 2)
```
Comment (eernstg) [inline LanguageFeatures/Records/primitive_equality_A01_t03.dart:37]:
```suggestion
      six: const Symbol('foo')): 2
```

File: LanguageFeatures/Records/primitive_equality_A01_t03.dart [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  };
+  const m5 = {
+    const (1, two: "2", String, four: C(), five: Object(), Symbol('foo')): 3
+  };
+  const m6 = {
+    (1, two: "2", String, four: const C(), five: const Object(),
+      const Symbol('foo'): 3)
```
Comment (eernstg) [inline LanguageFeatures/Records/primitive_equality_A01_t03.dart:44]:
```suggestion
      const Symbol('foo')): 3
```

File: LanguageFeatures/Records/primitive_equality_A01_t03.dart [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+///
+/// Note that this is a dynamic property of a record object, not a property of
+/// its static type. Since primitive equality only comes into play in constants,
+/// the compiler can see the actual field values for a relevant record at
+/// compile time because it has the actual constant record value with all of its
+/// constant fields. This means a record can be used in a constant set or as a
+/// constant map key, but only when its field values could be as well.
+///
+/// @description Checks that if all of the record fields have a primitive ==
+/// operator then this record can be used in a constant set
```
Comment (eernstg) [inline LanguageFeatures/Records/primitive_equality_A01_t03.dart:16]: Maybe?:
```suggestion
/// operator then this record can be used in a constant map literal as a key
```

**Comment by sgrekhov** [comment]
Thank you. Please rereview

**Review by eernstg — APPROVED** [review]
LGTM

**Pull request merged**